src/protocol.rs

```diff
@@ -499,7 +499,12 @@ pub enum LocalProtocolViolation {
     /// See [`ChannelConfiguration::with_max_request_payload_size()`] and
     /// [`ChannelConfiguration::with_max_response_payload_size()`] for details.
     #[error("payload exceeds configured limit")]
-    PayloadExceedsLimit,
+    PayloadExceedsLimit {
```

Comment:
c1da2026d09d5d11018a3f1b3eb281605832933f I'm not sure I'm a big fan of this, as it seems halfway. The change greatly increases the size of the error `enum`, which isn't too bad, since it rarely used (local protocol violations are completely avoidable).

On the other hand, if we're adding this, the other variants could use the same treatment.

Reply:
As discussed on Slack, I'll update the other variants, and add the info to the display impl.

Reply:
Done in [1b98b8c](https://github.com/casper-network/juliet/pull/7/commits/1b98b8cb8f17b8b023045395ec6cc2146dfc5acb).

Reply:
For the record, having `channel_count` as a struct member makes the hairs at the back of my neck stand up a bit, since it is a compile-time value. But we'll go with the discussed solution for now :)